src/client/proxy/middleware/cors.ts

```diff
@@ -0,0 +1,18 @@
+import {preHandlerHookHandler} from 'fastify';
+
+/**
+ * Allows Cross-Origin Requests to `https://mirror.point`.
+ */
+export const cors: preHandlerHookHandler = (req, res, done) => {
+    const protocol = req.protocol;
+    const {host} = req.urlData();
+
+    if (`${protocol}://${host}` === 'https://mirror.point') {
+        // Requests that are a result of a redirection during a CORS request
+        // have the `Origin` header set to `null`.
+        // Hence, we allow all origins (*) for `https://mirror.point`.
```

Comment:
This is intentional behaviour:

> If a cross-origin resource redirects to another resource at a new origin, the browser will set the value of the Origin header to null after redirecting. This prevents additional confused deputy attacks [...]

If you don't want the wildcard, please let me know and I'll see what we can do about it. The only thing that comes to mind is not redirecting, but perhaps there are other solutions I'm not aware of.